freetypefont.mod/freetypefont.bmx

```diff
@@ -166,7 +166,41 @@ Type TFreeTypeFont Extends BRL.Font.TFont
 				
 		Local ft_face:Byte Ptr
 
-		If src.Find( "::" )>0
+		If TStream(src)
+			Local stream:TStream = TStream(src)
+			Local data:Byte[1024]
+			Local dataSize:Int
+
+			'backup old stream position to set it back to it afterwards
+			Local oldStreamPos:int = stream.Pos()
+			stream.Seek(0)
+			
+			While Not stream.Eof()
+				If dataSize = data.length
+					data = data[.. dataSize*3/2]
+				EndIf
+				dataSize :+ stream.Read( (Byte Ptr data) + dataSize, data.length - dataSize )
+			Wend
+			If dataSize <> data.length
+				data = data[..dataSize]
+			EndIf
+
+			'restore stream position
+			stream.Seek(oldStreamPos)
+
+			'load data into freetype buffer
+			buf_size = data.length
+			If Not buf_size
+				Return Null
+			EndIf
+			buf = MemAlloc( Size_T(buf_size) )

```

Comment:
I'm wondering, since we already have a buffer (byte array) at this point, whether we may as  well just use that, and not allocate another pointer with a copy of it? (which we then need to clean up properly at some point).
This would allow us to remove the MemFree calls for buf, and have it clean itself when the font is collected.

Reply:
"buf" is used later on (stored in the freetype-font-instance).

but yes, I thought so too (to just use a pointer to the `tmp:Byte[]`-array) but thought it would be better for you if the code behaved similar to the "incbin"-part of the function (right below).

Reply:
yes, I meant buf can probably be replaced with a Byte array, which is then managed by the GC.

Reply:
Sounds more and more as if Mr. BaH accepts this mission :-)